tensorflow/c/eager/c_api_internal.h

```diff
@@ -96,13 +100,24 @@ struct TFE_TensorDebugInfo {
   std::vector<tensorflow::int64> dev_dims;
 };
 
+struct TFE_OpInferenceContext {
+  explicit TFE_OpInferenceContext(const tensorflow::OpDef* op_def)
+      : op_def(op_def) {}
+
+  const tensorflow::OpDef* op_def;  // op definition from protobuf
+  int input_idx = 0;                // index of the next input to be added
+  std::set<std::string> attrs;      // attributes inferred so far
```

Comment:
nit: This is touched multiple times on every op execution. use absl::flat_hash_set instead.